rpcs3/Emu/RSX/RSXFragmentProgram.h

```diff
@@ -265,7 +266,8 @@ struct RSXFragmentProgram
 				data_ptr = other.data_ptr;
 				local_storage.clear();
 			}
-
+			
+			assert(this != &other);
```

Comment:
```suggestion
			ensure(this != &other);
```

Reply:
This is a copy operator. This check is pointless.

Reply:
lol. didn't even notice

Reply:
it's a protection check against self assignment

Reply:
Self assignment is a valid statement, just return *this on it and it's dne.

Reply:
the function below it protects against weird memory issues by wrapping the whole thing in `if (this != &other)`; maybe that would be better?

Reply:
Yes.